lib/core/unzipomatic.ts

```diff
@@ -0,0 +1,36 @@
+import {Readable} from "stream";
+
+export type UnzipOptions = {
+    // ToDo, but I would start with making yauzl defaults non-configureable and see if there is demand for flexibility there
+}
+
+export type SourceType = string | File | Blob | Buffer | Readable
+
+export type TargetFileMetadata = {
+    fullPath: string
+    fileSize: number
+}
+
+// ToDo
+export type FileGenerator = Generator<Entry, void, void>
+
+export function unzipToFilesystem(source: SourceType, targetDir: string, options: UnzipOptions): Promise<TargetFileMetadata>
+
+/**
+ * Used to iterate over multiple files in an archive
+ */
+export function unzipToReadableGenerator(source: SourceType, options: UnzipOptions): Promise<FileGenerator>
+
+/**
+ * Used to extract a single-file archive
+ */
+export function unzipToReadable(source: SourceType, options: UnzipOptions): Promise<Readable>
+
+/**
+ * Used to extract a single-file archive
+ */
+export function unzipToBuffer(source: SourceType, options: UnzipOptions): Promise<Buffer>
```

Comment:
Shall we aim for a web api version as well?

Reply:
are there real-world use-cases for it?

Reply:
wonder if that blocks us from using Rust dependencies

Reply:
When we get those requests, I think we will have a better understanding of the `unzip` code and then we can export something like `unzipomatic/browser` and expose some APIs just for the browser.

But until we have requests for that, I don't think we should worry and we should do everything to speedup on node.